src/Orleans/project.json

```diff
@@ -1,6 +1,7 @@
-{
+﻿{
   "dependencies": {
-    "Newtonsoft.Json": "7.0.1"
+    "Newtonsoft.Json": "7.0.1",
+    "System.Collections.Immutable": "1.1.36"
```

Comment:
Hmmm, I'm a little hesitant to add this dependency here, as every grain project will be coupled to this too, and there are many ways to get into conflicts, especially since Roslyn uses it too, which runs on many different environments.
Not pushing back on this yet, but can we find another option to avoid this extra dependency in the core?
IMO, Newtonsoft.Json should probably be removed from the core too, but before adding more just for the sake of serialization, let's see if we can think of alternatives. Otherwise by the same token we probably should be adding F# types that are immutable or several others that we know how to serialize.